### DISCUSSION TITLE
How install in ubuntu 18.0.4

### DISCUSSION THREAD

**Discussion by ricardo345** [postalserver/postal#1455] [Help with using Postal]
Help. the installation guide and script for ubuntu 16.0.4 is deprecated, how install in ubuntu 18.0.4?

**Comment by ArriMa** [comment]
Why not an install guide on Ubuntu 20.04.2.0 LTS?

**Comment by willpower232** [comment]
There is already an 18.04 script but I think it needs some tweaks to work fully from https://github.com/postalhq/postal/pull/1310

I'm not sure about 20.04 but it may use a ruby version that is too new out of the box so could be problematic as other issues have noted.

**Comment by thisisbari** [comment]
# Run these commands one by one and it will work on ubuntu 18.0.4
-------xx-------------
sudo apt update
sudo apt install mariadb-server libmysqlclient-dev
-------xx-------------
sudo systemctl stop mariadb.service
sudo systemctl start mariadb.service
sudo systemctl enable mariadb.service
-------xx-------------
sudo mysql_secure_installation

sudo mysql -u root -p

Exit;

-------xx-------------
sudo apt-get install software-properties-common
sudo apt-add-repository ppa:brightbox/ruby-ng

-------xx-------------
sudo apt update
sudo apt install ruby2.3 ruby2.3-dev build-essential

-------xx-------------
wget -O- https://packages.erlang-solutions.com/ubuntu/erlang_solutions.asc | sudo apt-key add -
echo "deb https://packages.erlang-solutions.com/ubuntu bionic contrib" | sudo tee /etc/apt/sources.list.d/rabbitmq.list
sudo apt-get update
sudo apt-get install erlang
sudo apt-get update

-------xx-------------
curl -s https://packagecloud.io/install/repositories/rabbitmq/rabbitmq-server/script.deb.sh | sudo bash
sudo apt-get install rabbitmq-server

-------xx-------------

sudo systemctl stop rabbitmq-server
sudo systemctl start rabbitmq-server
sudo systemctl enable rabbitmq-server

-------xx-------------

sudo rabbitmqctl add_vhost /postal
sudo rabbitmqctl add_user postal p0stalpassw0rd
sudo rabbitmqctl set_permissions -p /postal postal "." "." ".*"

-------xx-------------

sudo apt install nodejs
sudo apt install git

-------xx-------------
sudo mysql -u root -p 
CREATE DATABASE postal CHARSET utf8mb4 COLLATE utf8mb4_unicode_ci;

CREATE USER 'postal'@'127.0.0.1' IDENTIFIED BY 'p0stalpassw0rd';

GRANT ALL ON `postal`.* TO `postal`@`127.0.0.1` IDENTIFIED BY "p0stalpassw0rd";
GRANT ALL PRIVILEGES ON `postal-%` . * to `postal`@`127.0.0.1`  IDENTIFIED BY "p0stalpassw0rd";

FLUSH PRIVILEGES;
EXIT;


-------xx-------------
sudo useradd -r -m -d /opt/postal -s /bin/bash postal
sudo setcap 'cap_net_bind_service=+ep' /usr/bin/ruby2.3

-------xx-------------
sudo gem install bundler
sudo gem install procodile
sudo gem install nokogiri -v '1.7.2'

-------xx-------------
sudo mkdir -p /opt/postal/app
sudo wget https://postal.atech.media/packages/stable/latest.tgz
sudo tar xvf latest.tgz -C /opt/postal/app
sudo chown -R postal:postal /opt/postal
sudo ln -s /opt/postal/app/bin/postal /usr/bin/postal

-------xx-------------

cd /opt/postal/app
bundle update mimemagic
bundle install
cd ~
-------xx-------------

postal bundle /opt/postal/vendor/bundle
postal initialize-config
postal initialize

-------xx-------------
postal make-user
-------xx-------------
sudo -u postal postal start
sudo -u postal postal status
-------xx-------------
sudo apt install nginx
sudo cp /opt/postal/app/resource/nginx.cfg /etc/nginx/sites-available/default
-------xx-------------
sudo mkdir /etc/nginx/ssl/
sudo openssl req -x509 -newkey rsa:4096 -keyout /etc/nginx/ssl/postal.key -out /etc/nginx/ssl/postal.cert -days 365 -nodes


-------xx-------------
/etc/nginx/sites-available  default file
-------xx-------------
sudo systemctl stop nginx 
sudo systemctl start nginx
-------xx-------------
postal restart

**Reply by abdallahtaleb** [reply]
The package for postal in here:  https://postal.atech.media/packages/stable/latest.tgz
was removed. Any update about this?

**Reply by adamcooke** [reply]
This method of installation is no longer supported.

Refer to our docs at https://docs.postalserver.io for full instructions on how to install Postal.

**Reply by thisisbari** [reply]
well try replacing it with "sudo wget https://github.com/postalserver/postal.git"
and don't forget to replace the file name with the one you get by downloading through this link

**Reply by adamcooke** [reply]
> well try replacing it with "sudo wget https://github.com/postalserver/postal.git"

This won't work. It'll download the HTML page for the repo's homepage on GitHub.

Just go and read the new instructions and do what they say.

**Reply by thisisbari** [reply]
If there are new instruction I would recommend you guys check them out thanks for the heads up

**Comment by mdobydullah** [comment]
Thanks @thisisbari. It works perfectly.

**Comment by ArriMa** [comment]
thank you, I've reinstalled on 18.04
